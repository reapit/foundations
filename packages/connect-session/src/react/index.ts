import React from 'react'
import { ReapitConnectHook, ReapitConnectSession } from '../types'
import { ReapitConnectBrowserSession } from '../browser'

export const useReapitConnect = (reapitConnectBrowserSession: ReapitConnectBrowserSession): ReapitConnectHook => {
  const [connectSession, setConnectSession] = React.useState<ReapitConnectSession | null>(null)

  React.useEffect(() => {
    const connectGetSession = async () => {
      const session = await reapitConnectBrowserSession.connectSession()
      if (session) {
        setConnectSession(session)
      }
    }

    connectGetSession()
  }, [])

  const connectAuthorizeRedirect = React.useCallback(() => {
    reapitConnectBrowserSession.connectAuthorizeRedirect()
  }, [])

  const connectLoginRedirect = React.useCallback(() => {
    reapitConnectBrowserSession.connectLoginRedirect()
  }, [])

  const connectLogoutRedirect = React.useCallback(() => {
    reapitConnectBrowserSession.connectLogoutRedirect()
  }, [])

  const connectIsDesktop = reapitConnectBrowserSession.connectIsDesktop

  const connectHasSession = reapitConnectBrowserSession.connectHasSession

  return {
    connectSession,
    connectAuthorizeRedirect,
    connectLoginRedirect,
    connectLogoutRedirect,
    connectIsDesktop,
    connectHasSession,
  }
}
