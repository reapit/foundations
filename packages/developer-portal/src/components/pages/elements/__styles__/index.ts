import { css } from 'linaria'

export const iframeWrapper = css`
  height: 88.5vh;
  width: 100%;
`
