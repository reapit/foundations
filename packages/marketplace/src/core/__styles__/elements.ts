import { css } from 'linaria'
import { reapitMidblue } from './colors'

export const link = css`
  color: ${reapitMidblue};
  font-weight: bold;
`

export const hyperlinked = css`
  color: ${reapitMidblue};
  color: ${reapitMidblue};
  text-decoration: underline;
  font-weight: bold;
`
