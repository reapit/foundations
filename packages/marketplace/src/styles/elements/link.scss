@import '../../styles/base/colors.scss';

.link {
  color: $reapit-marketplace-link-color;
  font-weight: $reapit-marketplace-link-font-weight;
}
