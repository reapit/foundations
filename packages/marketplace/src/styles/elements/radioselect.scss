.radioselect {
  margin-top: 0.5em;
}
