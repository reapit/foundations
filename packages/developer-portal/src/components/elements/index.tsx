import {
  SmallText,
  Button,
  elHFull,
  elMb5,
  FlexContainer,
  Icon,
  PageContainer,
  SecondaryNavContainer,
  BodyText,
  Title,
} from '@reapit/elements'
import React, { FC } from 'react'
import { openNewPage, ExternalPages } from '../../utils/navigation'
import { iframeWrapper } from './__styles__/index'

export const ElementsPage: FC = () => {
  return (
    <FlexContainer isFlexAuto>
      <SecondaryNavContainer>
        <Icon className={elMb5} icon="elementsInfographic" iconSize="large" />
        <BodyText>Reapit Elements</BodyText>
        <SmallText hasGreyText>
          Elements is a UI toolkit built in React JS, exporting a library of CSS classes to help you build clean,
          interactive user interfaces for the Reapit Foundations ecosystem. It is based on the Reapit Foundations Design
          System, our in house styleguide.
        </SmallText>
        <Button className={elMb5} intent="default" onClick={openNewPage(ExternalPages.elementsDocs)}>
          View Docs
        </Button>
        <Button className={elMb5} intent="primary" onClick={openNewPage(process.env.elementsUri)}>
          Storybook
        </Button>
      </SecondaryNavContainer>
      <PageContainer className={elHFull}>
        <Title>Elements</Title>
        <iframe
          className={iframeWrapper}
          src={process.env.elementsUri}
          frameBorder="0"
          allow="clipboard-write"
        ></iframe>
      </PageContainer>
    </FlexContainer>
  )
}

export default ElementsPage
