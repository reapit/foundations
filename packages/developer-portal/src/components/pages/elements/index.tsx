import {
  BodyText,
  Button,
  elHFull,
  elMb3,
  elMb8,
  FlexContainer,
  Icon,
  PageContainer,
  SecondaryNav,
  SecondaryNavContainer,
  SecondaryNavItem,
  Subtitle,
  Title,
} from '@reapit/elements'
import React, { FC } from 'react'
import { useHistory, useLocation } from 'react-router'
import Routes from '../../../constants/routes'
import { openNewPage, ExternalPages, navigate } from '../../../utils/navigation'
import { iframeWrapper } from './__styles__/index'

export const ElementsPage: FC = () => {
  const location = useLocation()
  const history = useHistory()
  const { pathname } = location
  return (
    <FlexContainer isFlexAuto>
      <SecondaryNavContainer>
        <Title>Docs</Title>
        <SecondaryNav className={elMb8}>
          <SecondaryNavItem onClick={navigate(history, Routes.API_DOCS)} active={pathname === Routes.API_DOCS}>
            API Docs
          </SecondaryNavItem>
          <SecondaryNavItem onClick={navigate(history, Routes.ELEMENTS)} active={pathname === Routes.ELEMENTS}>
            Elements
          </SecondaryNavItem>
        </SecondaryNav>
        <Icon icon="elementsInfographic" iconSize="large" />
        <Subtitle>Reapit Elements</Subtitle>
        <BodyText hasGreyText>
          Elements is a UI toolkit built in React JS, exporting a library of CSS classes to help you build clean,
          interactive user interfaces for the Reapit Foundations ecosystem. It is based on the Reapit Foundations Design
          System, our in house styleguide.
        </BodyText>
        <Button className={elMb3} intent="neutral" onClick={openNewPage(ExternalPages.elementsDocs)}>
          View Docs
        </Button>
        <Button className={elMb3} intent="critical" onClick={openNewPage(window.reapit.config.elementsUri)}>
          Open New
        </Button>
      </SecondaryNavContainer>
      <PageContainer className={elHFull}>
        <Title>Elements</Title>
        <iframe
          className={iframeWrapper}
          src={window.reapit.config.elementsUri}
          frameBorder="0"
          allow="clipboard-write"
        ></iframe>
      </PageContainer>
    </FlexContainer>
  )
}

export default ElementsPage
