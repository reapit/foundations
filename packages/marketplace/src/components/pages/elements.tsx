import * as React from 'react'
import ErrorBoundary from '@/components/hocs/error-boundary'
import { FlexContainerBasic } from '@reapit/elements'
import { IFRAME_URLS } from '../../constants/iframe-urls'

const ElementsPage: React.SFC = () => (
  <ErrorBoundary>
    <FlexContainerBasic flexColumn hasPadding>
      <FlexContainerBasic className="container">
        <iframe
          style={{ border: 'none', height: '100%' }}
          scrolling="no"
          src={IFRAME_URLS.elements}
          width="100%"
          height="100%"
        />
      </FlexContainerBasic>
    </FlexContainerBasic>
  </ErrorBoundary>
)

export default ElementsPage
