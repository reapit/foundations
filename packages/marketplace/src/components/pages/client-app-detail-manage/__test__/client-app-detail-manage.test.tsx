import * as React from 'react'
import * as ReactRedux from 'react-redux'
import TestRenderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme'
import configureStore from 'redux-mock-store'
import { MemoryRouter } from 'react-router'
import ClientAppDetailManage, {
  handleCloseUninstallConfirmationModal,
  handleUninstallAppButtonClick,
  renderAppHeaderButtonGroup,
} from '../client-app-detail-manage'
import { Button } from '@reapit/elements'
import Routes from '@/constants/routes'
import appState from '@/reducers/__stubs__/app-state'

describe('ClientAppDetailManage', () => {
  let store
  beforeEach(() => {
    /* mocking store */
    const mockStore = configureStore()
    store = mockStore(appState)
  })
  it('should match a snapshot', () => {
    expect(
      mount(
        <ReactRedux.Provider store={store}>
          <MemoryRouter
            initialEntries={[{ pathname: Routes.CLIENT_APP_DETAIL_MANAGE, key: 'clientAppDetailManageRoute' }]}
          >
            <ClientAppDetailManage />
          </MemoryRouter>
        </ReactRedux.Provider>,
      ),
    ).toMatchSnapshot()
  })

  describe('renderAppHeaderButtonGroup', () => {
    const mockInstalledOn = '2020-2-20'
    it('should match snapshot', () => {
      const wrapper = shallow(<div>{renderAppHeaderButtonGroup(mockInstalledOn, jest.fn())}</div>)
      expect(wrapper).toMatchSnapshot()
    })
    it('should render install app button if installedOn is existed', () => {
      const testRenderer = TestRenderer.create(renderAppHeaderButtonGroup(mockInstalledOn, jest.fn()))
      const testInstance = testRenderer.root
      expect(testInstance.findByType(Button).props.children).toBe('Uninstall App')
    })
  })
  describe('handleCloseUninstallConfirmationModal', () => {
    const mockFunction = jest.fn()
    const fn = handleCloseUninstallConfirmationModal(mockFunction)
    fn()
    expect(mockFunction).toBeCalledWith(false)
  })
  describe('handleUninstallAppButtonClick', () => {
    const mockFunction = jest.fn()
    const fn = handleUninstallAppButtonClick(mockFunction)
    fn()
    expect(mockFunction).toBeCalledWith(true)
  })
})
