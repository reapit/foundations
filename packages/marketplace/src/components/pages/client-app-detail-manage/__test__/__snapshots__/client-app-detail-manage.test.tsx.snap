// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientAppDetailManage renderAppHeaderButtonGroup should match snapshot 1`] = `
<div>
  <Component
    dataTest="detail-modal-uninstall-button"
    onClick={[MockFunction]}
    type="button"
    variant="primary"
  >
    Uninstall App
  </Component>
</div>
`;

exports[`ClientAppDetailManage should match a snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "key": "clientAppDetailManageRoute",
          "pathname": "/client/apps/:appid/manage",
        },
      ]
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "clientAppDetailManageRoute",
              "pathname": "/client/apps/:appid/manage",
              "search": "",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "clientAppDetailManageRoute",
            "pathname": "/client/apps/:appid/manage",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <ClientAppDetailManage>
        <div>
          <AppHeader
            appDetailData={Object {}}
            buttonGroup={
              <React.Fragment>
                
              </React.Fragment>
            }
          >
            <Component
              className="is-vcentered "
            >
              <div
                className="columns  is-vcentered "
              >
                <Component
                  className="is-4"
                >
                  <div
                    className="column  is-4"
                  >
                    <div>
                      <img
                        className="image"
                        src="https://bulma.io/images/placeholders/48x48.png"
                      />
                    </div>
                  </div>
                </Component>
                <Component
                  className="is-8"
                >
                  <div
                    className="column  is-8"
                  >
                    <Component>
                      <h3
                        className="title is-3  "
                        id=""
                      />
                    </Component>
                  </div>
                </Component>
              </div>
            </Component>
          </AppHeader>
          <AppContent
            appDetailData={Object {}}
            loginType="CLIENT"
          >
            <Component>
              <div
                className="columns  "
              >
                <Component
                  className="is-4"
                >
                  <div
                    className="column  is-4"
                  />
                </Component>
                <Component
                  className="is-8"
                >
                  <div
                    className="column  is-8"
                  >
                    <Component
                      html=""
                    >
                      <Component
                        className="html-render"
                      >
                        <div
                          className="content html-render"
                        />
                      </Component>
                    </Component>
                    <br />
                    <br />
                    <Component
                      isMultiLine={true}
                    >
                      <div
                        className="columns is-multiline "
                      />
                    </Component>
                  </div>
                </Component>
              </div>
            </Component>
          </AppContent>
        </div>
      </ClientAppDetailManage>
    </Router>
  </MemoryRouter>
</Provider>
`;
